cpp/tests/groupby/sum_with_overflow_tests.cpp

```diff
@@ -0,0 +1,492 @@
+      children.push_back(overflow_col.release());
+      auto expect_vals = cudf::create_structs_hierarchy(4, std::move(children), 0, {});
+
+      auto agg = cudf::make_sum_with_overflow_aggregation<cudf::groupby_aggregation>();
+      test_single_agg(keys, vals, expect_keys, *expect_vals, std::move(agg));
+    } else {
+      // For unsigned types: only test positive overflow
+      cudf::test::fixed_width_column_wrapper<V> vals{
+        static_cast<V>(large_positive),   // Group 1: Close to max
+        static_cast<V>(small_val1),       // Group 2: Small value
+        static_cast<V>(small_val2),       // Group 3: Small value
+        static_cast<V>(small_val1),       // Group 4: Small value
+        static_cast<V>(small_increment),  // Group 1: Will cause positive overflow
+        static_cast<V>(small_val2),       // Group 2: Small value
+        static_cast<V>(small_val3),       // Group 2: Small value
+        static_cast<V>(large_positive),   // Group 1: Close to max (second occurrence)
+        static_cast<V>(large_positive),   // Group 3: Close to max
+        static_cast<V>(1),                // Group 3: Small value
+        static_cast<V>(small_val4),       // Group 2: Small value
+        static_cast<V>(small_val2),       // Group 4: Small value
+        static_cast<V>(small_val3)};      // Group 4: Small value
+
+      cudf::test::fixed_width_column_wrapper<K> expect_keys{1, 2, 3, 4};
+
+      // Expected results: Groups 1 and 3 overflow; Groups 2 and 4 do not
+      auto sum_col = cudf::test::fixed_width_column_wrapper<V>{
+        static_cast<V>(static_cast<DeviceType>(large_positive) + small_increment +
+                       static_cast<DeviceType>(large_positive)),  // Group 1: overflowed result
+        static_cast<V>(small_val1 + small_val2 + small_val3 + small_val4),  // Group 2: no overflow
+        static_cast<V>(static_cast<DeviceType>(small_val2) +
+                       static_cast<DeviceType>(large_positive) +
+                       static_cast<DeviceType>(1)),           // Group 3: overflowed result
+        static_cast<V>(small_val1 + small_val2 + small_val3)  // Group 4: no overflow
+      };
+      auto overflow_col = cudf::test::fixed_width_column_wrapper<bool>{true, false, true, false};
+      std::vector<std::unique_ptr<cudf::column>> children;
+      children.push_back(sum_col.release());
+      children.push_back(overflow_col.release());
+      auto expect_vals = cudf::create_structs_hierarchy(4, std::move(children), 0, {});
+
+      auto agg = cudf::make_sum_with_overflow_aggregation<cudf::groupby_aggregation>();
+      test_single_agg(keys, vals, expect_keys, *expect_vals, std::move(agg));
+    }
+
+    // Note: SUM_WITH_OVERFLOW only works with hash groupby, not sort groupby
+  }  // end else block for non-decimal types
+}
+
+// Test that SUM_WITH_OVERFLOW works with int32_t (which used to be unsupported)
+TEST(groupby_sum_with_overflow_compatibility_test, int32_support)
+{
+  using K = int32_t;
```

Comment:
`K` is already defined at the global space. 
Either define them in each specific case like here (preferred in my opinion) or remove the specific ones.